test/replication-luatest/gh_9943_promote_split_brain_test.lua

```diff
@@ -0,0 +1,124 @@
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_all(function(g)
+    g.rs = replica_set:new({})
+    local rs_id = g.rs.id
+    g.box_cfg = {
+        replication_timeout = 0.1,
+        replication_synchro_timeout = 86400,
+        election_mode = 'manual',
+        replication = {
+            server.build_listen_uri('server1', rs_id),
+            server.build_listen_uri('server2', rs_id),
+            server.build_listen_uri('server3', rs_id),
+        },
+    }
+
+    for i = 1, 3 do
+        g.rs:build_and_add_server{
+            alias = 'server' .. i,
+            box_cfg = g.box_cfg,
+        }
+    end
+
+    g.rs:start()
+    g.rs:wait_for_fullmesh()
+    g.rs.servers[1]:exec(function() box.ctl.promote() end)
+    g.rs.servers[1]:wait_for_election_leader()
+    g.rs.servers[1]:exec(function()
+        box.schema.create_space('test', {is_sync = true}):create_index('pk')
+    end)
+    g.rs.servers[2]:wait_for_vclock_of(g.rs.servers[1])
+    g.rs.servers[3]:wait_for_vclock_of(g.rs.servers[1])
+end)
+
+g.after_all(function(g)
+    g.rs:stop()
+end)
+
+local function set_replication(instance, new_one)
+    instance:exec(function(repl)
+        box.cfg{replication = repl}
+    end, {new_one})
+end
+
+local function promote(instance)
+    instance:exec(function() box.ctl.promote() end)
+    instance:wait_for_election_leader()
+end
+
+local function test_promote_split_brain(g, with_data)
+    local new_replication = {
+        g.rs.servers[1].net_box_uri,
+        g.rs.servers[2].net_box_uri,
+    }
+
+    set_replication(g.rs.servers[3], {})
+    set_replication(g.rs.servers[1], new_replication)
+    set_replication(g.rs.servers[2], new_replication)
+
+    g.rs.servers[3]:exec(function()
+        box.cfg{replication_synchro_quorum = 1}
+    end)
+
+    promote(g.rs.servers[2])
+    g.rs.servers[1]:exec(function(id)
+        -- Without this wait PROMOTE may be not executed and exit with TimedOut
+        -- error. Let's make sure, that server_1 known about new leader.
+        t.helpers.retrying({}, function()
+            t.assert_equals(box.info.election.leader, id)
+        end)
+    end, {g.rs.servers[2]:get_instance_id()})
+    promote(g.rs.servers[1])
+    if with_data then
+        g.rs.servers[1]:exec(function()
+            box.space.test:replace({1, 's1'})
+        end)
+    end
+
+    promote(g.rs.servers[3])
+    t.assert(g.rs.servers[3]:get_synchro_queue_term() <
+             g.rs.servers[1]:get_synchro_queue_term())
+    if with_data then
+        g.rs.servers[3]:exec(function()
+            box.space.test:replace({1, 's3'})
+        end)
+    end
+
+    -- Split-Brain should be noticed.
+    set_replication(g.rs.servers[1], {
+        g.rs.servers[1].net_box_uri,
+        g.rs.servers[2].net_box_uri,
+        g.rs.servers[3].net_box_uri,
+    })
+
+    t.helpers.retrying({}, function()
+        t.assert_not_equals(g.rs.servers[1]:grep_log('ER_SPLIT_BRAIN'), nil)
+    end)
```

Comment:
Can you instead get it from the `info.replication` from upstream/downstream status? Log greps is the last resort.

Reply:
Sure, it's better to check `box.info.replication.upstream`

Reply:
Resolving since @Gerold103 approved the PR.